src/current.rs

```diff
@@ -0,0 +1,34 @@
+//! Information about the current thread and current Shuttle execution.
+//!
+//! This module provides access to information about the current Shuttle execution. It is useful for
+//! building tools that need to exploit Shuttle's total ordering of concurrent operations; for
+//! example, a tool that wants to check linearizability might want access to a global timestamp for
+//! events, which the [`context_switches`] function provides.
+
+use crate::runtime::execution::ExecutionState;
+use crate::runtime::task::clock::VectorClock;
+use crate::runtime::task::TaskId;
+
+/// The number of context switches that happened so far in the current Shuttle execution.
+///
+/// Note that this is the number of *possible* context switches, i.e., including times when the
+/// scheduler decided to continue with the same task. This means the result can be used as a
+/// timestamp for atomic actions during an execution.
+///
+/// Panics if called outside of a Shuttle execution.
+pub fn context_switches() -> usize {
+    ExecutionState::context_switches()
+}
+
+/// Get the current thread's vector clock
+pub fn clock() -> VectorClock {
+    crate::runtime::execution::ExecutionState::with(|state| {
+        let me = state.current();
+        state.get_clock(me.id()).clone()
+    })
+}
+
+/// Gets the clock for the thread with the given task ID
+pub fn clock_for(task_id: TaskId) -> VectorClock {
```

Comment:
Is this function being used anywhere?

Reply:
I don't think so, but I don't think it's doing any harm being there.